### DISCUSSION TITLE
Set File Path Dynamically

### DISCUSSION THREAD

**Discussion by apple162883** [nextgenhealthcare/connect#5499] [Q&A] [answered]
Hi 
Mirth Version 3.12.0.    Source -- File Reader    Destination ---  Database Writer

I am trying to read files from SMB path where we have folders based on the current Year. In the file path contains year in the form of

      smb://Username:Password/MirthTest/2022/Test. 

So all the files for 2022 will come in this folder and for 2023 will be in another folder. The only difference between the path will be the current year.
Is there a way where I can insert this year value dynamically in the Source tab

**Comment by pacmano1** [comment, marked as answer]
Not sure on the objective here but, sure you can set a channel global map var in the deploy script and/or rotate it on poll completion.  The path names accespts velocity syntax, but your "test read" button won't work.
